sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ResourceProviderDataTests.cs

```diff
@@ -83,5 +83,76 @@ public void ValidateCapitalizationIsIgnored()
             Assert.AreEqual(File.ReadAllText(TestData.GetLocation("ResourceProviderData/ResourceProviderData-TwoSpaces.json")).TrimEnd(), stjResult);
 #endif
         }
+
+        [Test]
+        public void ProxySerialization()
+        {
+            var options = new ModelReaderWriterOptions("J");
+            options.AddProxy(new ResourceProviderDataProxy());
+
+            var model = ModelReaderWriter.Read<ResourceProviderData>(BinaryData.FromString(WirePayload));
+            Assert.NotNull(model);
+            var binaryData = ModelReaderWriter.Write(model!, options);
+            Assert.AreEqual(File.ReadAllText(TestData.GetLocation("ResourceProviderData/ResourceProviderData-Collapsed-MissingId.json")).TrimEnd(), binaryData.ToString());
+            Assert.AreNotEqual(WirePayload, binaryData.ToString());
+
+            var model2 = ModelReaderWriter.Read<ResourceProviderData>(binaryData, options);
+            Assert.NotNull(model2);
+            Assert.NotNull(model2!.Id);
+            Assert.AreEqual("TestValue", model2.Id);
```

Comment:
It might be helpful to add a comment here to say that if the model was written without the proxy, this value would be different ... or whatever is the correct way of interpreting these tests so that maintainers can understand the intended behavior of the type.